### PR TITLE
docs: document Metrics

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Closes #

- [ ] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Comment by NGA-TRAN** [comment]
@alamb @carols10cents  @e-dard @tustvold @crepererum @mkmik @jacobmarble 
These are metrics I believe you have created. Can you help review them?
I use a few of these metrics to create I[Ox Metrics dashboard](https://us-west1-1.gcp.cloud2.influxdata.com/orgs/2f5ec5d98a8d6526/dashboards/07f20d557feae000?lower=now%28%29%20-%203h) that I think will be useful for performance measurement as well as discovering issues

File: docs/metrics.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+### Chunks
+| Metric name |  Code Name | Description |
+| --- | --- | --- |
+| catalog_chunks_mem_usage_bytes | memory_metrics | Total memory usage by chunks (MUB, RUB, OS statistics) |
+| catalog_loaded_chunks | chunk_storage | Total number of chunks (MUB, RUB, RUBandOS) for each table |
+| catalog_loaded_rows | row_count | Total number of rows (MUB, RUB, RUBandOS) for each table |
+| catalog_lock_total | ? | ? |
+| catalog_lock_wait_seconds_total | ? | ? |
+| ? | table_lock_tracker| ? |
```
Comment (tustvold) [inline docs/metrics.md:34]: I don't think this is a metric? It should being reported under `catalog_lock_total` and `catalog_lock_wait_seconds_total`.

These are metrics on the number of each type of lock acquired, and the total time spent blocked on acquiring them. They were added in #1355

File: docs/metrics.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+| ? | partition_lock_tracker | ? |
+| ? | chunk_lock_tracker | ? |
+| ? | timestamp_histogram| ? |
```
Comment (tustvold) [inline docs/metrics.md:37]: This is a breakdown of the timestamp distribution. It is only published for tables contained in a comma separate environment variable `INFLUXDB_IOX_ROW_TIMESTAMP_METRICS`.

It was added in #2101 and the environment flag was added in #2135

File: docs/data_management.md [hunk 3]
```diff
@@ -89,13 +89,19 @@ From drawings/data_lifecycle.monopic
                  └─────────────────────┘   └───────────────────────┘    └────────────────────┘
                             │                                                      │
                             │                                                      │
-         ┌──────────────────┼───────────────────┐                      ┌───────────┴────────────┐
```
Comment (tustvold) [inline docs/data_management.md:92]: This section probably needs a more concrete update in light of the Kafka-ification, the segment store no longer exists, but out of scope for this PR

**Review by tustvold — APPROVED** [review]
Thank you for doing this @NGA-TRAN , I've added comments on the metrics that I'm familiar with, mainly those under the Chunks heading. Hopefully we can keep this document up to date as we add/remove metrics

File: docs/metrics.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+| catalog_chunks_total | chunk_state | Number of chunks in different states (open, closed, compacting, compacted, writing_os, rub_and_os, ...)) |
```
Comment (tustvold) [inline docs/metrics.md:38]: FWIW this metric is probably going to be removed - it is superceded by catalog_loaded_chunks which is also easier to interpret

**Pull request merged**